ui/src/components/head/inline/loadApp.js

```diff
@@ -0,0 +1,59 @@
+// the app is bundled without any polyfills. babel transpiles the syntax
+// to es5 but we do not provide any polyfills for missing methods.
+// for that, we use polyfill.io.
+//
+// since that's a possbile point of failure, we have a checked-in copy of
+// the pf.io response for *all* polyfills that we may serve to everyone
+// as a worst-case back up. This fallback will be loaded if the PolyfillIO
+// switch is off.
+//
+// in that situation, they're gated, meaning they won't patch if they don't
+// need to, but they're still downloaded (currently ~20 kB gzipped) and
+// the file is still parsed
+// this is a global that's called at the bottom of the pf.io response,
+// once the polyfills have run. This may be useful for debugging.
+function guardianPolyfilled() {
+    try {
+        window.guardian.polyfilled = true;
+    } catch (e) {};
+}
+
+// Load the app and try to patch the env with polyfill.io
+// Adapted from https://www.html5rocks.com/en/tutorials/speed/script-loading/#toc-aggressive-optimisation
+(function(document, window) {
+    var src;
+    var script;
+    var pendingScripts = [];
+    var firstScript = document.scripts[0];
+    var scripts = [
+        window.guardian.polyfillioUrl,
+        window.guardian.bundleUrl
```

Comment:
Worth grouping these urls under `window.guardian.config.scripts` or something to keep the `guardian` object tidier? (for instance, stylesheet URLs are currently found under ``window.guardian.config.stylesheets`)